consensus/src/epoch_manager.rs

```diff
@@ -563,27 +563,58 @@ impl<P: OnChainConfigProvider> EpochManager<P> {
         block_store: Arc<BlockStore>,
         max_blocks_allowed: u64,
     ) {
-        let (request_tx, mut request_rx) =
-            aptos_channel::new::<_, DeprecatedIncomingBlockRetrievalRequest>(
-                QueueStyle::KLAST,
-                10,
-                Some(&counters::BLOCK_RETRIEVAL_TASK_MSGS),
-            );
+        let (request_tx, mut request_rx) = aptos_channel::new::<_, IncomingBlockRetrievalRequest>(
+            QueueStyle::KLAST,
+            10,
+            Some(&counters::BLOCK_RETRIEVAL_TASK_MSGS),
+        );
         let task = async move {
             info!(epoch = epoch, "Block retrieval task starts");
             while let Some(request) = request_rx.next().await {
-                if request.req.num_blocks() > max_blocks_allowed {
-                    warn!(
-                        "Ignore block retrieval with too many blocks: {}",
-                        request.req.num_blocks()
-                    );
-                    continue;
-                }
-                if let Err(e) = monitor!(
-                    "process_block_retrieval",
-                    block_store.process_block_retrieval(request).await
-                ) {
-                    warn!(epoch = epoch, error = ?e, kind = error_kind(&e));
+                match request.req {
+                    // TODO @bchocho @hariria deprecate once BlockRetrievalRequest enum release is complete
+                    BlockRetrievalRequest::V1(v1) => {
+                        if v1.num_blocks() > max_blocks_allowed {
+                            warn!(
+                                "Ignore block retrieval with too many blocks: {}",
+                                v1.num_blocks()
+                            );
+                            continue;
+                        }
+                        if let Err(e) = monitor!(
+                            "process_block_retrieval",
+                            block_store
+                                .process_block_retrieval(DeprecatedIncomingBlockRetrievalRequest {
+                                    req: v1,
+                                    protocol: request.protocol,
+                                    response_sender: request.response_sender,
+                                })
+                                .await
+                        ) {
+                            warn!(epoch = epoch, error = ?e, kind = error_kind(&e));
+                        }
+                    },
+                    BlockRetrievalRequest::V2(v2) => {
+                        if v2.num_blocks() > max_blocks_allowed {
+                            warn!(
+                                "Ignore block retrieval with too many blocks: {}",
+                                v2.num_blocks()
+                            );
+                            continue;
+                        }
+                        if let Err(e) = monitor!(
+                            "process_block_retrieval",
```

Comment:
Change to `process_block_retrieval_v2` so we can see the change in metrics while rolling out

Reply:
Done!